Update to the gRPC overview doc for full beta release (#22605)
MystenLabs/sui: docs/content/guides/operator/sui-full-node.mdx

A Sui Full node is essentially a read-only view of the network state. Unlike validator nodes, Full nodes cannot sign transactions, although they can validate the integrity of the chain by re-executing transactions that a quorum of validators previously committed.

> Today, a Sui Full node has the potential to maintain the full history of the chain. But that will change with gRPC and as we phase out JSON-RPC.

Reviewer: ```suggestion
Today, a Sui Full node has the potential to maintain the full history of the chain. That will change with gRPC and as JSON-RPC gets phased out.
```

cmake
```

### Considerations to enable gRPC (beta)

To serve the [gRPC API in beta](/concepts/grpc-overview.mdx), you must enable it on your Full node by indexing gRPC-specific data. During the initial gRPC indexing phase, your Full node may not be able to handle other traffic — including the [JSON-RPC](/references/sui-api.mdx) requests. Plan your rollout carefully (such as using the `rolling upgrade` mechanism), and communicate any downtime to your customers and partners in advance. The time required to sync gRPC indexes depends on your Full node's hardware and software configuration.

> Enable gRPC indexing on your Full node using the following configuration:

Reviewer: ```suggestion
Enable gRPC indexing on your Full node by adding the following entry to your fullnode.yaml configuration:
```

```sh

rpc:
  enable-indexing: true

```

> Running gRPC alongside JSON-RPC will increase your Full node’s storage usage. While this is expected in the short term, you can reclaim this storage in the future by disabling JSON-RPC after it is phased out. Developers must gradually migrate to gRPC or [GraphQL RPC](/concepts/graphql-rpc.mdx) (which will internally rely on gRPC). You can also run gRPC and JSON-RPC on separate nodes to optimize resource usage and provide a performant plus resilient RPC experience to applications. Use the following config to selectively disable JSON-RPC on a Full node:

Reviewer: ```suggestion
Running gRPC alongside JSON-RPC increases your Full node’s storage usage. While this is expected in the short term, you can reclaim this storage in the future by disabling JSON-RPC after it is phased out. You must gradually migrate to gRPC or [GraphQL RPC](/concepts/graphql-rpc.mdx) (which internally relies on gRPC). You can also run gRPC and JSON-RPC on separate nodes to optimize resource usage and provide a performant and resilient RPC experience to applications. Add the following to your fullnode.yaml configuration to selectively disable JSON-RPC on a Full node:
```

Author: @ronny-mysten - I'm taking bits of this. Basically the migration comment has to be focused on "Developers" as this doc is for full node operator config and "You" may not be compatible. Also, GraphQL doesn't use gRPC yet but will in future, hence "will rely" for now is appropriate.

```sh

enable-index-processing: false

```

> You can configure the retention window for the gRPC index data on a Full node using the following config:

Reviewer: ```suggestion
You can configure the retention window for the gRPC index data on a Full node by adding the following to your fullnode.yaml configuration:
```

```sh

authority-store-pruning-config:
  num-epochs-to-retain: 14
  num-epochs-to-retain-for-checkpoints: 14

```

> Adjust the gRPC data retention period as needed, based on your Full node’s resource profile and depending on if you've disabled JSON-RPC or not. In any case, be sure to test the longer retention period for performance and scalability before using gRPC in your application or offering it to other developers.

Reviewer: ```suggestion
Adjust the gRPC data retention period based on your Full node’s resource profile and whether you've disabled JSON-RPC or not. In any case, test the longer retention period for performance and scalability before using gRPC in your application or offering it to other developers.
```

> We strongly recommend enabling gRPC beta support on your Full nodes as early as possible to help developers evaluate and integrate the API before general availability. Refer to [Sui's data serving roadmap](/guides/developer/getting-started/data-serving.mdx#grpc-api) for guidance on the upcoming transition from JSON-RPC.

Reviewer: ```suggestion
It is strongly recommended that you enable gRPC beta support on your Full nodes as early as possible to help developers evaluate and integrate the API before general availability. Refer to [Sui's data serving roadmap](/guides/developer/getting-started/data-serving.mdx#grpc-api) for guidance on the upcoming transition from JSON-RPC.
```
